iter/forEach_test.go

```diff
@@ -0,0 +1,26 @@
+package iter_test
+
+import (
+	"github.com/BooleanCat/go-functional/internal/assert"
```

Comment:
I should find a way to have the linter enforce this but I like imports separated so built-ins are first, then this project:

```go
import (
    "testing"

    "github.com/BooleanCat/go-functional/internal/assert"
    "github.com/BooleanCat/go-functional/iter"
)
```